Developers Guide.md

```diff
@@ -354,6 +354,33 @@ Do not:
   ```
 * Do not use `pragma`, and try to avoid `__attribute__` as well.
 
+### Documentation
+Documentation is done in Doxygen comments format, which must be put in header files.
+Exception are static or not exposed to the public API functions - they should be documented in 
```

Comment:
As per our voting result, please remove "or not exposed to the public API functions". You can use instead "not having declaration"  or "having only definition" (generally I mean those which are not in .h for some reason).